test_kubeyaml.py

```diff
@@ -26,7 +37,7 @@ def strip(s):
 port_numbers = strats.integers(min_value=1, max_value=32767)
 
 hostnames_without_port = strats.builds('.'.join,
-    strats.lists(elements=host_components, min_size=1, max_size=6))
+    strats.lists(elements=host_components, min_size=2, max_size=6))
```

Comment:
On the face of it, this seems like it will reduce coverage -- a hostname can have a single component. Was it necessary for some reason?

Reply:
_Thinks_

If I recall correctly, it generated a `0/a:a` output, which got mapped as `image: '0/a', tag: 'a'`.

Reply:
The reason I changed this is because with the generated image parts there is a chance we generate an image ref with: one host component, one image name component and a tag.

```
AssertionError: assert {'_containers...egistry': '0'} == {'_containers'...registry': ''}
E                 Omitting 1 identical items, use -vv to show
E                 Differing items:
E                 {'registry': '0'} != {'registry': ''}
E                 {'image': 'a:a'} != {'image': '0/a:a'}
E                 Use -v to get the full diff
```

There is however a limitation in the Docker image ref spec, which is also visible in `docker` itself, e.g. `docker pull hostname/a:a` will attempt to resolve `hostname/a` from the configured default registry instead of consulting 'hostname' (this is probably the reason they have a hardcoded `localhost` in their regex).

```
❯ docker pull hostname/a:a
Error response from daemon: pull access denied for hostname/a, repository does not exist or may require 'docker login'
❯ docker pull localhost/a:a
Error response from daemon: Get http://localhost/v2/: dial tcp [::1]:80: connect: connection refused
```

I will change it so that we use image names with at least two parts in places where we also make use of a registry.

Reply:
Got it, OK.